atlasdb-commons/src/main/java/com/palantir/common/proxy/InterruptibleProxy.java

```diff
@@ -47,7 +47,9 @@
 
     private final Object delegate;
     private static final ExecutorService executor = PTExecutors.newCachedThreadPool(
-            new NamedThreadFactory("Interruptible Proxy", true /* isDaemon */ ));
+            new NamedThreadFactory("Interruptible Proxy", true /* isDaemon */ ),
+            // Timeout idle threads after 5 seconds, up from the default of 100ms
+            5000);
```

Comment:
Chatted with @cakeofny this morning, might want to just inject executor in constructor as it simplifies the proxy and lets user control lifecycle of executor (right now nothing shuts this down)

Reply:
Separate from this PR we might want to look at executors across AtlasDB as 100ms default TTL for cached pools is quite short in general